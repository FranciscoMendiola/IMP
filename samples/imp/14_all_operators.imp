# aqui hay un comentario lo deberia ignorar
suma := a + b;
resta := a - b;
mult := a * b;


igual := a = b;
menor := a <= b;


conj := true and false;
neg := not true
