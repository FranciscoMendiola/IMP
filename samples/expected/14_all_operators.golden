Token {tokenKind = Comment, lexeme = "# aqui hay un comentario lo deberia ignorar", position = 0}
Token {tokenKind = Id, lexeme = "suma", position = 44}
Token {tokenKind = Assign, lexeme = ":=", position = 49}
Token {tokenKind = Id, lexeme = "a", position = 52}
Token {tokenKind = Plus, lexeme = "+", position = 54}
Token {tokenKind = Id, lexeme = "b", position = 56}
Token {tokenKind = Semi, lexeme = ";", position = 57}
Token {tokenKind = Id, lexeme = "resta", position = 59}
Token {tokenKind = Assign, lexeme = ":=", position = 65}
Token {tokenKind = Id, lexeme = "a", position = 68}
Token {tokenKind = Minus, lexeme = "-", position = 70}
Token {tokenKind = Id, lexeme = "b", position = 72}
Token {tokenKind = Semi, lexeme = ";", position = 73}
Token {tokenKind = Id, lexeme = "mult", position = 75}
Token {tokenKind = Assign, lexeme = ":=", position = 80}
Token {tokenKind = Id, lexeme = "a", position = 83}
Token {tokenKind = Times, lexeme = "*", position = 85}
Token {tokenKind = Id, lexeme = "b", position = 87}
Token {tokenKind = Semi, lexeme = ";", position = 88}
Token {tokenKind = Id, lexeme = "igual", position = 92}
Token {tokenKind = Assign, lexeme = ":=", position = 98}
Token {tokenKind = Id, lexeme = "a", position = 101}
Token {tokenKind = Eq, lexeme = "=", position = 103}
Token {tokenKind = Id, lexeme = "b", position = 105}
Token {tokenKind = Semi, lexeme = ";", position = 106}
Token {tokenKind = Id, lexeme = "menor", position = 108}
Token {tokenKind = Assign, lexeme = ":=", position = 114}
Token {tokenKind = Id, lexeme = "a", position = 117}
Token {tokenKind = Leq, lexeme = "<=", position = 119}
Token {tokenKind = Id, lexeme = "b", position = 122}
Token {tokenKind = Semi, lexeme = ";", position = 123}
Token {tokenKind = Id, lexeme = "conj", position = 127}
Token {tokenKind = Assign, lexeme = ":=", position = 132}
Token {tokenKind = KTrue, lexeme = "true", position = 135}
Token {tokenKind = KAnd, lexeme = "and", position = 140}
Token {tokenKind = KFalse, lexeme = "false", position = 144}
Token {tokenKind = Semi, lexeme = ";", position = 149}
Token {tokenKind = Id, lexeme = "neg", position = 151}
Token {tokenKind = Assign, lexeme = ":=", position = 155}
Token {tokenKind = KNot, lexeme = "not", position = 158}
Token {tokenKind = KTrue, lexeme = "true", position = 162}
